src/components/web-form-renderer.vue

```diff
@@ -0,0 +1,336 @@
+      <p v-else>
+        {{ $t(submissionModal.type + '.body') }}
+      </p>
+      <div class="modal-actions">
+        <button type="button" class="btn btn-primary" @click="hideSubmissionModal()">
+          {{ $t('action.close') }}
+        </button>
+      </div>
+    </template>
+  </modal>
+</template>
+
+<script setup>
+import { computed, createApp, getCurrentInstance, reactive } from 'vue';
+import { useRoute } from 'vue-router';
+/* eslint-disable-next-line import/no-unresolved -- not sure why eslint is complaining about it */
+import { OdkWebForm, webFormsPlugin } from '@getodk/web-forms';
+import useForm from '../request-data/form';
+import { apiPaths, isProblem } from '../util/request';
+import Loading from './loading.vue';
+import Modal from './modal.vue';
+import { modalData } from '../util/reactivity';
+import useRequest from '../composables/request';
+import { useRequestData } from '../request-data';
+import { noop } from '../util/util';
+
+const { resourceStates } = useRequestData();
+const { form, formVersionXml } = useForm();
+const { request } = useRequest();
+const submissionModal = modalData();
+const attachmentUploads = reactive([]);
+const primaryInstance = reactive({ instanceId: null, uploadSuccess: null, file: null });
+const route = useRoute();
+
+defineOptions({
+  name: 'WebFormRenderer'
+});
+
+const props = defineProps({
+  actionType: String
+});
+
+// Install WebFormsPlugin in the component instead of installing it at the
+// application level so that @getodk/web-forms package is not loaded for every
+// page, thus increasing the initial bundle
+const app = createApp({});
+app.use(webFormsPlugin);
+const inst = getCurrentInstance();
+// webFormsPlugin just adds globalProperty ($primevue)
+inst.appContext.config.globalProperties = {
+  ...inst.appContext.config.globalProperties,
+  ...app._context.config.globalProperties
+};
+
+const { initiallyLoading } = resourceStates([form, formVersionXml]);
+
+const isPublicLink = computed(() => !!route.query.st);
+
+const fetchData = () => {
+  let url = apiPaths.formXml(form.projectId, form.xmlFormId, !!form.draftToken);
+  if (form.draftToken) {
+    url = url.replace(/^\/v1\//, `/v1/test/${form.draftToken}/`);
+  }
+  if (route.query.st) {
+    url += `?st=${route.query.st}`;
+  }
+  formVersionXml.request({
+    url
+  });
+};
+
+fetchData();
+
+const showModal = (type) => {
+  submissionModal.state = true;
+  submissionModal.type = `${type}Modal`;
+};
+
+const hideSubmissionModal = () => {
+  submissionModal.hide();
+  primaryInstance.instanceId = null;
+};
+
+/**
+ * Web Form expects host application to provide a function that it can use to
+ * fetch attachments. Signature of the function is (url) => Response; where
+ * Response is subset of web standard  {@link Response}.
+ */
+const getAttachment = (url) => {
+  let requestUrl = apiPaths.formAttachment(
+    form.projectId,
+    form.xmlFormId,
+    form.draftToken !== null,
+    url.pathname.substring(1)
```

Comment:
Before, this was `url.pathname.split('/').pop()`, which is different from `url.pathname.substring(1)`. Did something change in the `url` that Web Forms emits?